component/src/main/java/io/siddhi/extension/store/mongodb/MongoSelectExpressionVisitor.java

```diff
@@ -0,0 +1,307 @@
+    }
+
+    @Override
+    public void endVisitMathRightOperand(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void beginVisitOr() {
+    }
+
+    @Override
+    public void endVisitOr() {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$or:" + "[" + leftOperand + "," + rightOperand + "]}");
+    }
+
+    @Override
+    public void beginVisitOrLeftOperand() {
+    }
+
+    @Override
+    public void endVisitOrLeftOperand() {
+    }
+
+    @Override
+    public void beginVisitOrRightOperand() {
+    }
+
+    @Override
+    public void endVisitOrRightOperand() {
+    }
+
+    @Override
+    public void beginVisitAnd() {
+    }
+
+    @Override
+    public void endVisitAnd() {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$and:" + "[" + leftOperand + "," + rightOperand + "]}");
+    }
+
+    @Override
+    public void beginVisitAndLeftOperand() {
+    }
+
+    @Override
+    public void endVisitAndLeftOperand() {
+    }
+
+    @Override
+    public void beginVisitAndRightOperand() {
+    }
+
+    @Override
+    public void endVisitAndRightOperand() {
+    }
+
+    @Override
+    public void beginVisitIsNull(String streamId) {
+        this.isNullCheck = true;
+    }
+
+    @Override
+    public void endVisitIsNull(String streamId) {
+        this.isNullCheck = false;
+        String nullCheckAttribute = conditionOperands.pop();
+        conditionOperands.push("{ $cond: { if: { $eq: [" + nullCheckAttribute + ", null ] }, then: true, " +
+                "else: false } }");
+    }
+
+    @Override
+    public void beginVisitCompare(Compare.Operator operator) {
+    }
+
+    @Override
+    public void endVisitCompare(Compare.Operator operator) {
+        String compareFilter;
+        switch (operator) {
+            case EQUAL:
+                compareFilter = "$eq";
+                break;
+            case GREATER_THAN:
+                compareFilter = "$gt";
+                break;
+            case GREATER_THAN_EQUAL:
+                compareFilter = "$gte";
+                break;
+            case LESS_THAN:
+                compareFilter = "$lt";
+                break;
+            case LESS_THAN_EQUAL:
+                compareFilter = "$lte";
+                break;
+            case NOT_EQUAL:
+                compareFilter = "$ne";
+                break;
+            default:
+                throw new MongoTableException("MongoDB Event Table found unknown operator '" + operator + "' for " +
+                        "COMPARE operation.");
+        }
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{ $cond: { if: { " + compareFilter + ": [" + leftOperand + "," + rightOperand + " ] }, then: true," +
```

Comment:
Reuse https://github.com/siddhi-io/siddhi-store-mongodb/blob/master/component/src/main/java/io/siddhi/extension/store/mongodb/util/MongoTableConstants.java#L54 also do the same check as in https://github.com/siddhi-io/siddhi-store-mongodb/blob/master/component/src/main/java/io/siddhi/extension/store/mongodb/MongoExpressionVisitor.java#L215  . Applicable to other methods

Reply:
1. Cannot reuse since their order of operands & operator is different. Created a new variable.
https://github.com/siddhi-io/siddhi-store-mongodb/pull/56/commits/94fba768238670966474ed6ac8a757455790895b#

Reply:
2. In SelectionExpressionVisitor do I need to do the conditional checks? Current implementation can support for multiple expressions as in mongodb.

Reply:
Then it's fine. Let's not add that